src/VisualStudio/CSharp/Test/ProjectSystemShim/CPS/CSharpCompilerOptionsTests.cs

```diff
@@ -102,32 +102,22 @@ public async Task ProjectOutputBinPathChange_CPS()
 
         [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
         [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
-        public async Task InvalidProjectOutputBinPaths_CPS1()
+        public async Task InvalidProjectOutputBinPaths_CPS()
         {
             using var environment = new TestEnvironment();
-            using var project1 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test", binOutputPath: null);
-            // Null output path is allowed.
-            Assert.Null(project1.BinOutputPath);
-        }
-
-        [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
-        [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
-        public async Task InvalidProjectOutputBinPaths_CPS2()
-        {
-            using var environment = new TestEnvironment();
-            using var project2 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: String.Empty);
-            // Empty output path is not allowed, it gets reset to null.
-            Assert.Null(project2.BinOutputPath);
+            await Assert.ThrowsAsync<InvalidOperationException>(() => CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: null));
+            await Assert.ThrowsAsync<InvalidOperationException>(() => CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test2", binOutputPath: String.Empty));
         }
 
         [WpfFact, WorkItem(14520, "https://github.com/dotnet/roslyn/issues/14520")]
         [Trait(Traits.Feature, Traits.Features.ProjectSystemShims)]
         public async Task InvalidProjectOutputBinPaths_CPS3()
         {
             using var environment = new TestEnvironment();
-            using var project3 = await CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test3", binOutputPath: "Test.dll");
-            // Non-rooted output path is not allowed, it gets reset to a temp rooted path.
-            Assert.Equal(Path.Combine(Path.GetTempPath(), "Test.dll"), project3.BinOutputPath);
+
+            // Non-rooted output path is not allowed, the project system will handle the exception:
+            await Assert.ThrowsAsync<InvalidOperationException>(() =>
+                CSharpHelpers.CreateCSharpCPSProjectAsync(environment, "Test3", binOutputPath: "Test.dll"));
```

Comment:
This should probably look at the exception returned from ThrowsAsync and assert that the message is what we expect (or at least close to that) so that way an unrelated bug can't creep in for this case and it's throwing somewhere else. I could imagine a mistake here might result in say passing this to a file system function that'd also throw InvalidOperationException due to the incorrect path form.

Reply:
Changed the exception type and testing for that type.